src/vaadin-number-field.html

```diff
@@ -344,15 +344,16 @@
           this.validate();
         }
 
-        validate() {
+        checkValidity() {
```

Comment:
Can be overridden as, not sure which one is easier to read 🤔:
```
checkValidity() {
  // text-field mixin does not check against `min`, `max` and `step`
  if (this.min !== undefined || this.max !== undefined || this.__validateByStep) {
    return this.inputElement.checkValidity();
  }
  
  return super.checkValidity;
}

Reply:
The shorter version you suggested looks nicer, not a blocker